src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1362,11 +1357,42 @@ private SslClientAuthenticationOptions GetSslOptionsForRequest(HttpRequestMessag
             return _sslOptionsHttp11!;
         }
 
-        private HttpConnection ConstructHttp11Connection(Stream stream, TransportContext? transportContext)
+        private async ValueTask<Stream> ApplyPlaintextFilter(Stream stream, Version httpVersion, HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (Settings._plaintextStreamFilter is null)
+            {
+                return stream;
+            }
+
+            try
+            {
+                return await Settings._plaintextStreamFilter(new SocketsHttpPlaintextStreamFilterContext(stream, httpVersion, request), cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                throw new HttpRequestException(SR.net_http_exception_during_plaintext_filter, e);
```

Comment:
What happens if we don't wrap this; does the non-HttpRequestException escape to user code?  I'm wondering if we really need this try/catch.

If we do keep the try/catch, should it be `catch (Exception e) when (e is not HttpRequestException)`?

Reply:
Yes, it escapes to user code.

I don't think it's worth adding the when clause to check for HttpRequestException. The expectation here is that this callback should never throw. If it does, it seems best to provide an explicit exception for this, even if the inner exception is HttpRequestException.

Reply:
Ok